tests/components/kodi/test_binary_sensor.py

```diff
@@ -0,0 +1,78 @@
+"""The tests for Kodi binary sensor platform."""
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import async_update_entity
+
+from . import init_integration
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "element, name",
+    [("screensaver", "Screensaver"), ("energy_saving", "Energy saving")],
+)
+async def test_binary_sensor_initial_states(hass: HomeAssistant, element, name):
+    """Test for binary sensor values."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_OFF
+    assert state.attributes["friendly_name"] == f"{entry.data['name']} {name}"
+
+
+@pytest.mark.parametrize("element", ["screensaver", "energy_saving"])
+async def test_binary_sensor_statechange(hass: HomeAssistant, element):
+    """Test the event."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_OFF
+
+    hass.states.async_set(f"binary_sensor.{entry.data['name']}_{element}", STATE_ON)
+    await hass.async_block_till_done()
```

Comment:
don't you also want to assert that the relevant kodi call was made?

Reply:
You mean something like this:
```
assert len(calls) == 1
assert calls[0].data["some"] == f"turn_on - {kodi_media_player} - 0"
```
to be honest, I haven't understood this. Is this for the home assistant internal signalling?
I will look at it.

Reply:
You patch the relevant call in the kodi library, and then use it to assert that it was called and with what parameters.

Reply:
Ah, sorry I made a mistake. There is no call to Kodi from HA. This was my first test of changing the state but you cannot actually set the screensaver state. They are only updated from Kodi (either poll or push).
I will replace this with the web socket update you already suggested.

Reply:
Yeah, this is a binary sensor. Setting its state from HA is not really something we should test.